R/summarise_country.R

```diff
@@ -0,0 +1,163 @@
+summarise_sheet_indicator_list <- function(billion) {
+  billion <- rlang::arg_match(billion)
+  indicator_order_b <- indicator_order %>%
+    dplyr::filter(ind %in% billion_ind_codes(billion)) %>%
+    dplyr::arrange(order) %>%
+    dplyr::select(sdg, short_name, medium_name, unit_raw, )
+
+  return(indicator_order_b)
+}
+
+#' Summarise Excel sheet 'data' for HPOP country summary
+#'
+#' `summarise_HPOP_cntry_data` summarises the 'data' sgeet for the HPOP country summary.
+#'
+#' @param iso3 ISO3 codes of country to export.
+#' @param ... Additional arguments passed to `calculate_hpop_contributions` and
+#' `calculate_hpop_billion`. Use if you need to provide additional parameters to
+#' those functions.
+#' @inherit calculate_hpop_contributions params
+#'
+#' @return list of data frames to be used in the 'data' sheet of HPOP country summary
+#' export
+#'
+#' @export
+summarise_HPOP_cntry_data <-
+  function(df,
+           iso3,
+           start_year = 2018,
+           end_year = 2019:2023,
+           ...) {
+    assert_mart_columns(df)
+    assert_years(start_year, end_year)
+
+    # Filter df for country, arrange indicators by order.
+    df_iso3 <- df %>%
+      dplyr::filter(iso3 == !!iso3) %>%
+      dplyr::left_join(dplyr::select(indicator_order, ind, order), by = "ind") %>%
+      dplyr::arrange(order) %>%
+      dplyr::select(-order)
+
+
+    #Get unique indicators
+    unique_ind <- unique(df_iso3[["ind"]])
+
+    # Data frame with indicators' order
+    ind_df <- indicator_order %>%
+      dplyr::filter(ind %in% unique_ind) %>%
+      dplyr::arrange(order) %>%
+      dplyr::select(ind, transformed_name, unit_transformed)
+
+    # Latest reported data
+    latest_reported <- df_iso3 %>%
+      transform_hpop_data() %>%
+      dplyr::group_by(iso3, ind) %>%
+      dplyr::filter(type %in% c("estimated", "reported")) %>%
+      dplyr::mutate(maxyear = max(year)) %>%
+      dplyr::filter(year >= maxyear) %>%
```

Comment:
Redundant `mutate()`, just put `year >= max(year)` and merge with previous `filter()`.